provider.go

```
@@ -77,6 +83,64 @@ func readJWTToken(path string) (string, error) {
 	return string(bytes.TrimSpace(data)), nil
 }
 
+func (p *Provider) getMountInfo(mountName, token string) (string, string, error) {
+	client, err := p.createHTTPClient()
+	if err != nil {
+		return "", "", err
+	}
+
+	addr := p.VaultAddress + "/v1/sys/mounts"
+	req, err := http.NewRequest(http.MethodGet, addr, nil)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "couldn't generate request")
+	}
+	// Set vault token.
+	req.Header.Set("X-Vault-Token", token)
+	req.Header.Set("X-Vault-Request", "true")
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "couldn't get sys mounts")
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		var b bytes.Buffer
+		io.Copy(&b, resp.Body)
+		return "", "", fmt.Errorf("failed to get successful response: %#v, %s",
+			resp, b.String())
+	}
+
+	var mount struct {
+		Data map[string]Mount `json:"data"`
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&mount); err != nil {
+		fmt.Printf("error: %s", err)
```

Should return the err here after logging